doc/AI-requirements.adoc

```diff
@@ -0,0 +1,170 @@
+For this project we will use agents, that will ask the neuronal network for a decision, when won't have continuous 
+decision making.
+
+== Communication Requirements
+
+This section will set the requirements for the communication between the three essentials parts _game code_, 
+_unity ml-agents_ und `BrainAcademy`.
+
+=== Game Code to Unity ML Agents
+
+This is probably the most import part as will have the most influence on if the AI is going to work. The _game code_ 
+not make any direct request or information sharing other than calling for a decision. This will kept the agents much more
+flexible, so it will be possible to changes agents quickly and develop different agents for different solutions.
+
+So therefor the _game code_ should provide information sharing methods. The `CollectObservation()` should be able to ask
+for a lot of different information. E.g "Give me all enemies." The cropping of the data should happen in the `BrainAcademy`.
+It should be possible to ask for everything. This way training speed would increase drastically as when won't need new 
+Unity builds, but could change it easily in the `BrainAcademy`. This means for an enemy it should provide a public accessor
+which wraps the enemy into a data object.
+
+=== Unity ML Agents to BrainAcademy
+
+One of the crucial parts of developing an neuronal network with reinforcement learning is the input it gets. In order
+to keep the size small and enable fast decision making we decided not use any visual observation. In addition to that
+the *input for ONE network must always have the same dimension* meaning the neuronal network needs have same choices every
+time no matter if this choice is valid or possible. So the _game code_ needs to validate the decision. So the neuronal
+network can learn during training, that this decision has no impact and will learn not use it.
+
+There is another approach which catches invalid decisions before hand, but this would only increase the data send from
+and forth between all the parts, so this might not be an option, but could be reconsider during training.
+****
+Agent::
+
+A decision for an agent is request with `Agent.RequestDecision()` as you can see it has no parameters. So in order to get
+the observation we need to implement a `CollectObservations(VectorSensor sensor)`. So has the Agent is a component of the
+NPC it will have access to other component.
+
+In addition to the decisions needs to have same dimension every time, the observation must have the same dimension as well.
+This could be tricky for example in mass battles where multiple allies and enemies are around. So in order to get this to
+work we need to cut the observation size to `n`-nearest persons and if are less around fill this vectors with zeros.
+
+To further reduce the size of the network and there for speed up the training process, the agent will have no memory.
```

Comment:
Are you talking about the agent from the BrainAcademy or the agent form ML_Agents?

If you are talking about the BrainAcademy agent:
This is only one option and no final decision. 
I definitely want to try agents with memory as well.

Reply:
Yes, I'm talking about the BrainAcademy's Agent. I think it would make no sense to implement a memory in Unity.

All right, I will at this, but my understanding was we don't want to try it. Maybe I missed it.